lib/elixometer.ex

```diff
@@ -332,19 +278,17 @@ defmodule Elixometer do
     :ok
   end
 
-  def handle_call({:subscribe, name}, _caller, state) do
-    if not metric_subscribed?(name) do
-      cfg = Application.get_all_env(:elixometer)
-      reporter = cfg[:reporter]
-      interval = cfg[:update_frequency]
-
-      if reporter do
-        :exometer.info(name)
-        |> Keyword.get(:datapoints)
-        |> Enum.map(&(:exometer_report.subscribe(reporter, name, &1, interval)))
-      end
-      :ets.insert(@elixometer_table, {{:subscriptions, name}, true})
+  def ensure_registered(name, register_fn) do
```

Comment:
This could use some documentation. It took me a minute to realize what "name" is here and how `subscribe(name)` (called within) differs from `subscribe_to(name)`.